tests/lib/rules/no-get-with-default.js

```diff
@@ -0,0 +1,34 @@
+const rule = require('../../../lib/rules/no-get-with-default');
+const RuleTester = require('eslint').RuleTester;
+
+const EXPECTED_ERROR_MESSAGE = 'Use `get` and `||` instead of `getWithDefault` for property `key`';
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2015,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('no-get-with-default', rule, {
+  valid: ["const test = this.get('key') || [];", "const test = get(this, 'target') || [];"],
```

Comment:
Need to add some more valid examples:

* `SomeRandomClass.getWithDefault('key', [])`
* `getWithDefault.someRandomFunction('key', [])`

Reply:
Good call on the false positives